cmd/metadata_l.go

```diff
@@ -0,0 +1,56 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"log"
+
+	"github.com/spf13/cobra"
+
+	"github.com/giantswarm/cloud-director-cli/pkg/vcd"
+	"github.com/giantswarm/cloud-director-cli/pkg/vcd/utils"
+)
+
+var (
+	href            string
+	listMetadataCmd = &cobra.Command{
+		Use:   "metadata",
+		Short: "List metadata of a resources such as vapp",
+		Long: `List metadata of a resources such as vapp
+
+	Example:
+	--------
+	cd-cli list metadata -href="https://mydomain.com/api/vApp/vapp-c7b89940-eece-48c7-8895-883fd347ee3e"
```

Comment:
so by default the metadata object is not fetched, but one can fetch it eventually having the href that's available in `.HREF` when doing `-ojson`, do I get it right? I am asking because we were talking about an option to populate those `.Metadata` objects when some switch is present for `cd-cli get ..`. But I like this approach even more actually, because it wouldn't be doing N additional http calls which can slow things down drastically.

Reply:
Yes. I did implement this way to reduce the number of https calls since there is no way to fetch metadata without sending a separate request.